app/src/main/java/ru/prometeydev/movie/model/MoviesRepository.kt

```diff
@@ -1,41 +1,52 @@
 package ru.prometeydev.movie.model
 
 import androidx.paging.*
-import kotlinx.coroutines.async
+import coil.annotation.ExperimentalCoilApi
+import kotlinx.coroutines.*
 import kotlinx.coroutines.flow.*
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
 import ru.prometeydev.movie.model.database.MoviesDatabase
 import ru.prometeydev.movie.model.database.entitiy.MoviesRemoteKeysEntity
 import ru.prometeydev.movie.model.domain.Genre
 import ru.prometeydev.movie.model.domain.Movie
 import ru.prometeydev.movie.model.mappers.*
 import ru.prometeydev.movie.model.network.MoviesApi
+import ru.prometeydev.movie.ui.MovieNotifications
 import ru.prometeydev.movie.ui.base.Result
 
 class MoviesRepository(
     private val api: MoviesApi,
-    private val db: MoviesDatabase
+    private val db: MoviesDatabase,
+    private val notifications: MovieNotifications
```

Comment:
I don't think that notifications should be called from the repository.

Reply:
From view model?

Reply:
I've tried to answer in the telegram channel